deb/build/debian/jenkins.init

```diff
@@ -43,6 +45,26 @@ if [ "$RUN_STANDALONE" = "false" ]; then
     exit 1
 fi
 
+# Make sure there exists a java executable, it may not be allways the case
+if [ -z "$JAVA" ]; then
+    echo "ERROR: No Java executable found in current PATH: $PATH" >&2
+    echo "If you actually have java installed on the system make sure the executable is in the aforementioned path and that 'which java' returns the java executable path" >&2
+    exit 1
+fi
+
+# Work out the JAVA version we are working with:
+JAVA_VERSION=$($JAVA -version 2>&1 | sed -n ';s/.* version "\(.*\)\.\(.*\)\..*"/\1\2/p;')
+
+if [ "$JAVA_VERSION" = "$JAVA_ALLOWED_VERSION" ]; then
```

Comment:
@oleg-nenashev Yes, it works perfectly with IBM java 8:

{noformat}
ubuntu@ubuntu-xenial:~$ java -version
java version "1.8.0"
Java(TM) SE Runtime Environment (build pxa6480sr4fp7-20170627_02(SR4 FP7))
IBM J9 VM (build 2.8, JRE 1.8.0 Linux amd64-64 Compressed References 20170616_352529 (JIT enabled, AOT enabled)
J9VM - R28_20170616_0201_B352529
JIT  - tr.r14.java_20170616_352529
GC   - R28_20170616_0201_B352529_CMPRSS
J9CL - 20170616_352529)
JCL - 20170622_01 based on Oracle jdk8u131-b11
ubuntu@ubuntu-xenial:~$ sudo service jenkins start
ubuntu@ubuntu-xenial:~$ sudo service jenkins status
● jenkins.service - LSB: Start Jenkins at boot time
   Loaded: loaded (/etc/init.d/jenkins; bad; vendor preset: enabled)
   Active: active (exited) since Mon 2017-07-10 10:22:39 UTC; 5s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 6285 ExecStop=/etc/init.d/jenkins stop (code=exited, status=0/SUCCESS)
  Process: 6386 ExecStart=/etc/init.d/jenkins start (code=exited, status=0/SUCCESS)

Jul 10 10:22:38 ubuntu-xenial systemd[1]: Starting LSB: Start Jenkins at boot time...
Jul 10 10:22:38 ubuntu-xenial jenkins[6386]: Correct java version found
Jul 10 10:22:38 ubuntu-xenial jenkins[6386]:  * Starting Jenkins Automation Server jenkins
Jul 10 10:22:38 ubuntu-xenial su[6427]: Successful su for jenkins by root
Jul 10 10:22:38 ubuntu-xenial su[6427]: + ??? root:jenkins
Jul 10 10:22:38 ubuntu-xenial su[6427]: pam_unix(su:session): session opened for user jenkins by (uid=0)
Jul 10 10:22:39 ubuntu-xenial jenkins[6386]:    ...done.
Jul 10 10:22:39 ubuntu-xenial systemd[1]: Started LSB: Start Jenkins at boot time.
{noformat}

Is IBM one of the supported JVM for jenkins? Not sure about this but, judging by the dependencies in the control file, IBM Java alone would not allow you to install the deb file. You would end having to install openjdk or the platform's default JVM which I suspect is not IBM on any case